commit ea7029410336675de8fecc7c6ebe62c8504ae988
Author: Alan Coopersmith <alan.coopersmith@sun.com>
Date:   Fri Jan 26 14:38:26 2007 -0800

    If using shadowfb, allocate shadow area before fbScreenInit
    
    (Copied from xf86-video-fbdev)

diff --git a/src/wsfb_driver.c b/src/wsfb_driver.c
index dcaa3c4..c6ad4d4 100644
--- a/src/wsfb_driver.c
+++ b/src/wsfb_driver.c
@@ -820,6 +820,17 @@ #endif
 
 	fPtr->fbstart = fPtr->fbmem;
 
+	if (fPtr->shadowFB) {
+	    fPtr->shadow = xcalloc(1, pScrn->virtualX * pScrn->virtualY *
+				   pScrn->bitsPerPixel);
+
+	    if (!fPtr->shadow) {
+		xf86DrvMsg(pScrn->scrnIndex, X_ERROR,
+			   "Failed to allocate shadow framebuffer\n");
+		return FALSE;
+	    }
+	}
+	
 	switch (pScrn->bitsPerPixel) {
 	case 1:
 		ret = xf1bppScreenInit(pScreen, fPtr->fbstart,
